src/sdk/util/passphrase.ts

```diff
@@ -0,0 +1,39 @@
+import { topPasswords } from '../topPassphrases';
+import { randomBytes } from './crypto';
+
+export function generateRandomKeywords(): string[] {
+    const randomIndices: number[] = [];
+    const maxIndex = topPasswords.length;
+
+    while (randomIndices.length < 6) {
+        // Generate a random 4-byte number
+        const randomBytesArray = randomBytes(4);
+        const randomIndex = byteArrayToNumber(randomBytesArray) % maxIndex;
+
+        if (!randomIndices.includes(randomIndex)) {
+            randomIndices.push(randomIndex);
+        }
+    }
+
+    const randomKeywords: string[] = randomIndices.map((index) => topPasswords[index]);
+
+    return randomKeywords;
+}
+
+export function generateAutoSuggestions(inputString: string): string[] {
+    if (inputString.trim() === '') {
+        return []; // Return an empty array for empty input
+    }
+
+    inputString = inputString.toLowerCase();
+    const matchingSuggestions: string[] = topPasswords
+        .filter((word: string) => word.toLowerCase().includes(inputString))
+        .slice(0, 4);
+
+    return matchingSuggestions;
+}
+
+// Convert a byte array to a number
+function byteArrayToNumber(byteArray: Uint8Array): number {
```

Comment:
As mentioned above, better to move something like this into our crypto.ts library

Reply:
["separation of conserns"](https://en.wikipedia.org/wiki/Separation_of_concerns)